docs/Account.md

```diff
@@ -0,0 +1,331 @@
+
+# 1. Deploy Account
+starknet = await Starknet.empty()
+account = await starknet.deploy(
+    "contracts/Account.cairo",
+    constructor_calldata=[signer.public_key]
+)
+
+# 2. Initialize Account
+await account.initialize(account.contract_address).invoke()
+
+# 3. Send transaction to Account
+await signer.send_transaction(account, account.contract_address, 'set_public_key', [other.public_key])
+```
+
+## Standard Interface
+
+The [`IAccount.cairo`](https://github.com/OpenZeppelin/cairo-contracts/blob/8739a1c2c28b1fe0b6ed7a10a66aa7171da41326/contracts/IAccount.cairo) contract interface contains the standard account interface proposed in [#41](https://github.com/OpenZeppelin/cairo-contracts/discussions/41) and adopted by OpenZeppelin and Argent. It implements [EIP-1271](https://eips.ethereum.org/EIPS/eip-1271) and it is agnostic of signature validation and nonce management strategies.
+
+```c#
+@contract_interface
+namespace IAccount:
+    #
+    # Getters
+    #
+
+    func get_nonce() -> (res : felt):
+    end
+
+    #
+    # Business logic
+    #
+
+    func is_valid_signature(
+            hash: felt,
+            signature_len: felt,
+            signature: felt*
+        ):
+    end
+
+    func execute(
+            to: felt,
+            selector: felt,
+            calldata_len: felt,
+            calldata: felt*,
+            nonce: felt
+        ) -> (response: felt):
+    end
+end
+```
+
+## Keys, signatures and signers
+
+While the interface is agnostic of signature validation schemes, this implementation assumes there's a public-private key pair controlling the Account. That's why the `constructor` function expects a `public_key` parameter to set it. Since there's also a `set_public_key()` method, accounts can be effectively transferred.
+
+Notice that although the current implementation works only with StarkKeys, support for Ethereum ECDSA algorithm is expected to be supported in the future.
+
+### Signer utility
+
+[Signer.py](https://github.com/OpenZeppelin/cairo-contracts/blob/8739a1c2c28b1fe0b6ed7a10a66aa7171da41326/tests/utils/Signer.py) is used to perform transactions on a given Account, crafting the tx and managing nonces.
+
+It exposes two functions:
+
+- `def sign(message_hash)` receives a hash and returns a signed message of it
+- `def send_transaction(account, to, selector_name, calldata, nonce=None)` returns a promise of a signed transaction, ready to be sent.
+
+To use Signer, pass a private key when instantiating the class:
+
+```python
+from utils.Signer import Signer
+
+signer = Signer(123456789987654321)
+```
+
+Then send transactions with `send_transaction` method.
+
+```python
+await signer.send_transaction(account, contract_address, 'method_name', [])
+```
+
+
+## Message format
+
+The idea is for all user intent to be encoded into a `Message` as follows:
+
+```c#
+struct Message:
+    member sender: felt
+    member to: felt
+    member selector: felt
+    member calldata: felt*
+    member calldata_size: felt
```

Comment:
This is not about documentation, but I was under the impression the convention was to do length first followed by the pointer. Like `signature_len, signature` and `calldata_len, calldata`. Doesn't this go against that convention? Also size vs len.

Reply:
discussed offline